### PR TITLE
fix : Alma client is not defined in BO

### PR DESCRIPTION
**Pull request by Francois-Gomis**

### CODE REVIEW

**Review by syjust-alma — APPROVED** [review]

**Review by syjust-alma — COMMENTED** [review]
I think the better way to manage this `InvalidArgumenException` will be to throw it directly from `getDefaultClient()` ...

As I can see, it could be used everywhere it needed by changing the __constructor assignation to `$this->alma = $almaClient` ... then, you will be able to use `$this->alma->getDefaultClient()->whatYouWant()` after and try catch on `InvalidArgumentException` where it is needed.

Finally, I think `InvalidArgumentException` could be replaced by a specific `AlmaDefaultClientException` or somthing like that 😉

**Review by syjust-alma — CHANGES_REQUESTED** [review]
I think the better way to manage this `InvalidArgumenException` will be to throw it directly from `getDefaultClient()` ...

As I can see, it could be used everywhere it needed by changing the __constructor assignation to `$this->alma = $almaClient` ... then, you will be able to use `$this->alma->getDefaultClient()->whatYouWant()` after and try catch on `InvalidArgumentException` where it is needed.

Finally, I think `InvalidArgumentException` could be replaced by a specific `AlmaDefaultClientException` or somthing like that 😉

File: Gateway/Config/PaymentPlans/PaymentPlansConfig.php [hunk 1]
```diff
@@ -82,13 +84,13 @@ public function __construct(
      */
     public function updateFromApi()
     {
-        $alma = $this->almaClient->getDefaultClient();
-        if (!$alma) {
-            $this->logger->info('Default client is not set - Impossible to update from Api',[]);
+        try {
+            $feePlans = $this->almaClient->getDefaultClient()->merchants->feePlans(FeePlan::KIND_GENERAL, "all", true);
+        } catch (RequestError | AlmaClientException $e) {
```
Comment (syjust-alma) [inline Gateway/Config/PaymentPlans/PaymentPlansConfig.php:89]: 😍

File: Gateway/Http/Client/RefundClient.php [hunk 2]
```diff
@@ -68,8 +70,16 @@ public function placeRequest(TransferInterface $transferObject): array
...
+            $fails = new stdClass();
+            $fails->responseCode = 460;
+            $fails->message = $e->getMessage();
+            return [
+                'resultCode' => 0,
+                'fails' => $fails,
+            ];
         } catch (RequestError $e) {
-            $this->logger->error("Error creating refund:", [$e]);
+            $this->logger->error("Error creating refund:", [$e->getMessage()]);
```
Comment (syjust-alma) [inline Gateway/Http/Client/RefundClient.php:82]: warning, sometime, getMessage is empty in `RequestError`

with [php-client 1.9.0](https://github.com/alma/alma-php-client/releases/tag/v1.9.0), you can now use `getErrorMessage`

**Review by syjust-alma — APPROVED** [review]

File: Gateway/Validator/RefundResponseValidator.php [hunk 3]
```diff
@@ -27,6 +28,9 @@ public function validate(array $validationSubject): ResultInterface
             if ($apiResponse->responseCode == '400') {
                 $fails[] = $apiResponse->json['error_code'] . ' : ' . $apiResponse->json['errors'][0]['message'];
             }
+            if ($apiResponse->responseCode == '460') {
```
[the remark this answers is not in the document]
Comment (Francois-Gomis) [reply Gateway/Validator/RefundResponseValidator.php:31]: I create custom response code for custom error message in backend

**Pull request merged**